gio/src/socket.rs

```diff
@@ -18,15 +18,18 @@ use crate::{ffi, Cancellable, Socket, SocketAddress, SocketControlMessage};
 impl Socket {
     #[cfg(unix)]
     #[cfg_attr(docsrs, doc(cfg(unix)))]
-    #[allow(clippy::missing_safety_doc)]
-    pub unsafe fn from_fd(fd: impl IntoRawFd) -> Result<Socket, glib::Error> {
+    #[doc(alias = "g_socket_new_from_fd")]
+    pub fn from_fd(fd: OwnedFd) -> Result<Socket, glib::Error> {
         let fd = fd.into_raw_fd();
         let mut error = ptr::null_mut();
-        let ret = ffi::g_socket_new_from_fd(fd, &mut error);
-        if error.is_null() {
-            Ok(from_glib_full(ret))
-        } else {
-            Err(from_glib_full(error))
+        unsafe {
+            let ret = ffi::g_socket_new_from_fd(fd, &mut error);
```

Comment:
Does this actually work? You'd pass the fd to the socket, but then the `OwnedFd` is dropped at the end of this scope and that would close the fd or not, leaving the socket with a closed fd.

Reply:
Same question also elsewhere, so let's figure this out before more review :)

Reply:
We use https://doc.rust-lang.org/std/os/fd/trait.IntoRawFd.html#tymethod.into_raw_fd right before this call. This should be fine.